### PR TITLE
Finish v0.11 migration

### PR DESCRIPTION
**Pull request by ethanfrey**
This finishes up the great work started in @maurolacy 's branch and makes a final cosmwasm-plus PR for the upgrades

### CODE REVIEW

**Comment by maurolacy** [comment]
Ey, thanks for finishing this! I'll review it today / tomorrow.

File: contracts/cw20-base/src/contract.rs [hunk 1]
```diff
@@ -902,7 +911,7 @@ mod tests {
         );
 
         // run the migration
-        let env = mock_env(HumanAddr::from("admin"), &[]);
+        let (env, _info) = mock_env_info(HumanAddr::from("admin"), &[]);
```
Comment (ethanfrey) [inline contracts/cw20-base/src/contract.rs:920]: don't we need to add info to migrate so it compiles in wasm mode?

Comment (orkunkl) [reply contracts/cw20-base/src/contract.rs:920]: Yes you are right. I did it locally but I suppose you are working on it don't want to tip toes.

**Comment by ethanfrey** [comment]
@maurolacy can you give a quick review on these changes before merging them into your branch (you should be able to do that). Then I can merge your branch into master. And we have 0.11 support!

I guess I wait til Monday to cut the v0.3.0 release

File: contracts/cw3-fixed-multisig/src/contract.rs [hunk 2]
```diff
@@ -339,15 +350,18 @@ fn reverse_proposals<S: Storage, A: Api, Q: Querier>(
...
-        .map(map_proposal)
+        .map(|p| map_proposal(&env.block, p))
         .collect();
 
     Ok(ProposalListResponse { proposals: props? })
 }
 
-fn map_proposal(item: StdResult<(Vec<u8>, Proposal)>) -> StdResult<ProposalResponse> {
+fn map_proposal(
+    block: &BlockInfo,
```
Comment (maurolacy) [inline contracts/cw3-fixed-multisig/src/contract.rs:360]: Nice.

File: contracts/cw3-fixed-multisig/src/contract.rs [hunk 3]
```diff
@@ -779,10 +794,10 @@ mod tests {
 
         // Expired proposals cannot be voted
         let env = match voting_period {
-            Duration::Time(duration) => mock_env_time(VOTER4, env.block.time + duration + 1),
-            Duration::Height(duration) => mock_env_height(VOTER4, env.block.height + duration + 1),
+            Duration::Time(duration) => mock_env_time(duration + 1),
+            Duration::Height(duration) => mock_env_height(duration + 1),
```
Comment (maurolacy) [inline contracts/cw3-fixed-multisig/src/contract.rs:798]: Nice.

File: contracts/cw3-fixed-multisig/src/state.rs [hunk 4]
```diff
@@ -32,12 +32,15 @@ pub struct Proposal {
...
-    pub fn current_status(&self) -> Status {
+    pub fn current_status(&self, block: &BlockInfo) -> Status {
         let mut status = self.status;
 
-        // if open, check if voting is passed on timed out
+        // if open, check if voting is passed or timed out
         if status == Status::Open && self.yes_weight >= self.required_weight {
-            status = Status::Passed
+            status = Status::Passed;
+        }
```
Comment (maurolacy) [inline contracts/cw3-fixed-multisig/src/state.rs:41]: Nice. No need to use `else if` here. Let's just not change the order of these checks.

File: contracts/cw721-base/src/contract.rs [hunk 5]
```diff
@@ -41,63 +42,62 @@ pub fn init<S: Storage, A: Api, Q: Querier>(
...
-    description: Option<String>,
-    image: Option<String>,
+    _env: Env,
+    info: MessageInfo,
+    msg: MintMsg,
+    // token_id: String,
+    // owner: HumanAddr,
+    // name: String,
+    // description: Option<String>,
+    // image: Option<String>,
```
Comment (maurolacy) [inline contracts/cw721-base/src/contract.rs:83]: Remove commented code.

File: contracts/cw721-base/src/contract.rs [hunk 6]
```diff
@@ -450,15 +463,22 @@ fn query_all_approvals<S: Storage, A: Api, Q: Querier>(
...
-        .map(|item| {
-            item.and_then(|(k, expires)| {
-                Ok(cw721::Approval {
-                    spender: deps.api.human_address(&k.into())?,
-                    expires,
-                })
-            })
+        .filter_map(|item| match item {
+            Ok((k, expires)) => {
+                // we remove all expired operators from the result
```
Comment (maurolacy) [inline contracts/cw721-base/src/contract.rs:468]: Sure about this? I'm not sure, but listing expired operators can be useful for debugging / accountability issues.

File: contracts/cw721-base/src/contract.rs [hunk 7]
```diff
@@ -497,9 +518,14 @@ fn query_all_nft_info<S: Storage, A: Api, Q: Querier>(
...
 
-fn humanize_approvals<A: Api>(api: A, info: &TokenInfo) -> StdResult<Vec<cw721::Approval>> {
+fn humanize_approvals<A: Api>(
+    api: A,
+    block: &BlockInfo,
+    info: &TokenInfo,
+) -> StdResult<Vec<cw721::Approval>> {
     info.approvals
         .iter()
+        .filter(|apr| !apr.expires.is_expired(block))
```
Comment (maurolacy) [inline contracts/cw721-base/src/contract.rs:528]: Same here.

**Review by maurolacy — APPROVED** [review]
Quick review, LGTM.

File: contracts/cw721-base/src/contract.rs (hunk 6 above)
Comment (ethanfrey) [reply contracts/cw721-base/src/contract.rs:468]: I guess I can keep them, but then add another flag to the response "is_expired: bool". Is it useful info on the client, without forcing calculations there

Comment (maurolacy) [reply contracts/cw721-base/src/contract.rs:468]: Good idea. Or directly add the expiration info.

Another option would be to add a boolean flag to the query, i. e. `include_expired` or so.

Comment (ethanfrey) [reply contracts/cw721-base/src/contract.rs]: Should be addressed here: https://github.com/CosmWasm/cosmwasm-plus/pull/111/commits/09b7b31427eb9c484e070697c91dba89c1a22310

Comment (ethanfrey) [reply contracts/cw721-base/src/contract.rs]: > Another option would be to add a boolean flag to the query, i. e. include_expired or so.

Just read this. I actually prefer this idea

Comment (ethanfrey) [reply contracts/cw721-base/src/contract.rs]: I will undo my change and redo it the other way. Let me know which commit you prefer.

Comment (maurolacy) [reply contracts/cw721-base/src/contract.rs]: I too think the query option is better.

Comment (ethanfrey) [reply contracts/cw721-base/src/contract.rs]: Look at https://github.com/CosmWasm/cosmwasm-plus/pull/111/commits/0525a341b7bc29cddaa40405c84ad4561aef6029

**Pull request merged**